sqlmesh/core/dialect.py

```diff
@@ -314,6 +314,31 @@ def _parse_join(
     return macro
 
 
+def _parse_select(
+    self: Parser,
+    nested: bool = False,
+    table: bool = False,
+    parse_subquery_alias: bool = True,
+    parse_set_operation: bool = True,
+) -> t.Optional[exp.Expression]:
+    select = self.__parse_select(  # type: ignore
+        nested=nested,
+        table=table,
+        parse_subquery_alias=parse_subquery_alias,
+        parse_set_operation=parse_set_operation,
+    )
+
+    if (
+        not select
+        and not parse_set_operation
+        and self._match_pair(TokenType.PARAMETER, TokenType.VAR, advance=False)
+    ):
+        self._advance()
+        return _parse_macro(self)
+
+    return select
```

Comment:
Let's try to simplify this, maybe this works too:

```python
    return select or self._parse_placeholder()
```

Reply:
This was my original implementation but it broke several tests, e.g:

```
E   SELECT *
E   FROM @this_model
E   WHERE @AND(@column NOT IN @is_in, @condition)
```

So I tried to shield it off such that we attempt to parse macros only during set operations.

Reply:
I see, that's ok then. Perhaps being explicit by checking the `PARAMETER` and `VAR` pair is better to avoid parsing edge cases. Please disregard my suggestion 👍